config/element-api.php

```diff
@@ -217,11 +217,148 @@ function getLegacyPage($locale)
+        // see https://craftcms.stackexchange.com/a/3013/6844 for `lft` param
+        $relatedEntry['order'] = $relatedItem->lft;
+        $relatedEntries[] = $relatedEntry;
+    }
+    return $relatedEntries;
+}
+
+function parseSegmentMatrix($entry, $locale)
+{
+    $segments = [];
+    if ($entry->segment) {
+        
+        foreach ($entry->segment->all() as $block) {
+            $segment = [];
+            $segment['title'] = $block->segmentTitle;
+            $segment['content'] = $block->segmentContent;
+            $segment['photo'] = $block->segmentImage->find()[0]->url;
+            array_push($segments, $segment);
+        }
+    }
+    return $segments;
+}
+
+function getFundingGuidancePage($locale)
+{
+    normaliseCacheHeaders(300);
+
+    $pagePath = \Craft::$app->request->getParam('path');
+
+    $searchCriteria = [
+        'section' => 'fundingGuidance'
+    ];
+
+    if ($pagePath) {
+        $searchCriteria['uri'] = $pagePath;
+    } else {
+        $searchCriteria['level'] = 1;
+    }
+    
+    return [
+        'serializer' => 'jsonApi',
+        'elementType' => Entry::class,
+        'criteria' => $searchCriteria,
+        'transformer' => function (Entry $entry) use ($locale, $pagePath) {
+            
+            $entryData = getBasicEntryData($entry);
+            // make a copy of this entry before augmenting it
+            //  so we can add it as a sibling
+            $thisEntry = $entryData;
+
+            if ($entry->mainHeading) {
+                $entryData['mainHeading'] = $entry->mainHeading;
+            }
+
+            if ($entry->photo) {
+                $photos = [];
+                foreach ($entry->photo->all() as $photo) {
+                    $photos[] = $photo->url;
+                }
+                if ($photos) {
+                    $entryData['photo'] = $photos[0];
+                }
+            }
+            
+            if ($entry->introductionText) {
+                $entryData['introduction'] = $entry->introductionText;
+            }
+
+            // @TODO this still returns true if the field is empty
+            if ($entry->outroText) {
+                $entryData['outro'] = $entry->outroText;
+            }
+
+            $segments = parseSegmentMatrix($entry, $locale);
+            if ($segments) {
+                $entryData['segments'] = $segments;
+            }
+            
+
+            $children = getRelatedEntries($entry, 'children');
+            if (count($children) > 0) {
+                $entryData['children'] = $children;
+            }
+            
+            $siblings = getRelatedEntries($entry, 'siblings');
+            if (count($siblings) > 0) {
+                $thisEntry['isCurrent'] = true;
+                $thisEntry['order'] = $entry->lft;
+                $entryData['siblings'] = $siblings;
+                $entryData['siblings'][] = $thisEntry;
+                // reorder the siblings now we've appended the current page
```

Comment:
This feels a little magic, anyway we could avoid the ordering? Does getting the parent first and then using `getDescendants` work?